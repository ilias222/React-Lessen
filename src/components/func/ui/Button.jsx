import {useState, useEffect, memo} from 'react'
import '../../../css/style.css'

export function Button(props){
    useEffect(() => {
        console.log("Ui did mounted")
        
        }, [props.setList])
    return(
        <div className='bar_form'>
        <input type="text" placeholder="Name" onInput={props.setName} className="bar_form__input-autor"/>
        <div class="flex-win">
        <textarea name="user_up__innerText" id="innerText" cols="30" rows="10" placeholder="Go to text"
        onChange={props.setText} className="bar_form__text-user">
        </textarea>
        <button onClick={props.setUp} className="bar_form__button-down">UP</button>
        </div>
        </div>
    )
}