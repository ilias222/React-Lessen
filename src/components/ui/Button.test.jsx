import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom'

import { Button } from "./Button"
import userEvent from "@testing-library/user-event";

describe('testing Button', () => {
    it('render component Button', () => {
        render(<Button>Change name</Button>)
    })

    it('render vith snapchot', () => {
        const {asFragment} = render(<Button>Change name</Button>)
        expect(asFragment()).toMatchSnapshot()
    })

    it('render vith snapchot', () => {
        render(<Button>Change name</Button>)
        expect(screen.getByText(/Change name/)).toBeInTheDocument()
    })

    // it('disable is Button', () => {
    //     render(<Button disable>Delete</Button>)
    //     expect(screen.getByText(/Delete/)).toBeDisabled()
    // })

    it('color is Button - black', () => {
        render(<Button>Delete</Button>)
        expect(screen.getByText(/Delete/)).toHaveStyle({color: "black"})
    })

    //  it('onclick to Button', async () => {
    //      const mochHandler = jest.fn()
    //      render(<Button onClick={mochHandler}>Deletes</Button>)
    //      await userEvent.click(screen.getByText(/Deletes/))
    //      expect(mochHandler).toHaveBeenCalledTimes(1)
    //  })

    it('test example', async () => {
        const onChange = jest.fn()
        render(<input type='checkbox' onChange={onChange}/>)
        const chekbox = screen.getByRole('checkbox')
        await userEvent.dblClick(chekbox)
        expect(onChange).toHaveBeenCalledTimes(2)
        expect(chekbox).not.toBeChecked()
    })

})