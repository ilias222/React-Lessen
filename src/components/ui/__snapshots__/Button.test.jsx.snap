// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Button render vith snapchot 1`] = `
<DocumentFragment>
  <button
    style="color: black;"
  >
    Change name
  </button>
</DocumentFragment>
`;
